src/impu_store.cpp

```diff
@@ -539,6 +539,40 @@ Store::Status ImpuStore::set_impu_without_cas(ImpuStore::Impu* impu,
   return status;
 }
 
+Store::Status ImpuStore::add_impu_without_cas(ImpuStore::Impu* impu,
+                                              SAS::TrailId trail)
+{
+  std::string data;
+
+  Store::Status status = impu->to_data(data);
+
+  if (status == Store::Status::OK)
+  {
+    int now = time(0);
+
+    // Set the data with a CAS of 0, which will fail if there's data already
+    // present
+    status = _store->set_data("impu",
+                              impu->impu,
+                              data,
+                              0,
+                              impu->expiry - now,
+                              trail,
+                              false);
+
+    if (status == Store::Status::DATA_CONTENTION)
```

Comment:
I don't think we should do this here. I think we should propagate this to the MemcachedCache, which can then make the decision that it's safe to ignore this.

Generally the ImpuStore doesn't have the knowledge to deal with data contention - that's the preserve of the MemcachedCache which has a higher level view.